examples/tree_3d_dgsem/elixir_mhd_dynamo.jl

```diff
@@ -0,0 +1,164 @@
+# A turbulent helically driven dynamo model in resistive compressible MHD.
+# We start with a weak small-scale magnetic field provided as Gaussian noise.
+# The energy is supplied via a forcing function f_force(x, t), which is delta-correlated in time.
+# For a comparison look into the Pencil Code example https://github.com/pencil-code/pencil-code/tree/master/samples/helical-MHDturb.
+
+using OrdinaryDiffEq
+using Trixi
+using Random
+using LinearAlgebra
+using StableRNGs
+
+###############################################################################
+# initial condition and forcing function
+
+"""
+    initial_condition_gaussian_noise_mhd(x, t, equations)
+
+Define the initial condition with a weak small-scale Gaussian noise magnetic field.
+"""
+function initial_condition_gaussian_noise_mhd(x, t, equations)
+    amplitude = 1e-8
+
+    seed = reinterpret(UInt, t + x[1] + x[2] + x[3])
+    rng = StableRNG(seed)
+
+    rho = 1.0
+    rho_v1 = 0.0
+    rho_v2 = 0.0
+    rho_v3 = 0.0
+    rho_e = 10.0
+    B1 = randn(rng) * amplitude
+    B2 = randn(rng) * amplitude
+    B3 = randn(rng) * amplitude
+    psi = 0.0
+
+    return SVector(rho, rho_v1, rho_v2, rho_v3, rho_e, B1, B2, B3, psi)
+end
+
+"""
+    source_terms_helical_forcing(u, x, t, equations::CompressibleMhdDiffusion3D)
+
+Forcing term that adds a helical small-scale driver to the system that is
+delta-correlated in time.
+"""
+function source_terms_helical_forcing(u, x, t, equations::IdealGlmMhdEquations3D)
+    # forcing amplitude
+    f_0 = 0.02
+    # helicality of the forcing (-1, 1)
+    sigma = 1.0
+
+    # Extract some parameters for the computation.
+    c_s = 0.1
+    delta_t = 0.01
+
+    # To make sure that the random numbers depend only on time we need to set the seeds.
+    seed = reinterpret(UInt, t)
+    rng = StableRNG(seed)
+
+    # Random phase -pi < phi <= pi
+    phi = (rand() * 2 - 1) * pi
+
+    # Random vector k, also delta-correlated in time.
+    k = [0.0, 0.0, 0.0]
+    k[1] = (randn(rng) * 2 - 1) * pi
+    k[2] = (randn(rng) * 2 - 1) * pi
+    k[3] = (randn(rng) * 2 - 1) * pi
+    k = k / norm(k)
+    k = 6 * k
+    k_norm = k / norm(k)
```

Comment:
That looks like an unnecessary allocation

Reply:
Done.